aws/cloudformation.yml

```diff
@@ -0,0 +1,699 @@
+
+                echo "script2" >> /var/tmp/bootstrap.txt
+                sudo -u mastodon bash -ex /var/tmp/stage2.sh  # Mastodon
+                echo "script 2 result $?" >> /var/tmp/bootstrap.txt
+                echo "-----" >> /var/tmp/bootstrap.txt
+
+                echo "script3" >> /var/tmp/bootstrap.txt
+                bash -ex /var/tmp/stage3.sh  # Nginx and Certbot
+                echo "script 3 result $?" >> /var/tmp/bootstrap.txt
+                echo "-----" >> /var/tmp/bootstrap.txt
+
+                echo "script4" >> /var/tmp/bootstrap.txt
+                sudo -u mastodon bash -ex /var/tmp/stage4.sh  # Crontab/Maintenance Jobs
+                echo "script 2 result $?" >> /var/tmp/bootstrap.txt
+                echo "-----" >> /var/tmp/bootstrap.txt
+
+                echo "Final reboot" >> /var/tmp/bootstrap.txt
+                systemctl reboot
+
+        create_files:
+          files: 
+            /var/tmp/stage1.sh:
+              content: !Sub |
+                #!/bin/bash -ex
+
+                echo "Create mastodon group" >> /var/tmp/bootstrap.txt
+                addgroup -gid 991 mastodon
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "Create mastodon user" >> /var/tmp/bootstrap.txt
+                adduser --quiet --disabled-password --gid 991 --uid 991 mastodon --gecos GECOS
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "Grant mastodon sudo" >> /var/tmp/bootstrap.txt
+                usermod -a -G sudo mastodon
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "change mastodon password" >> /var/tmp/bootstrap.txt
+                echo "mastodon:${MastodonLocalUserPassword}" | sudo chpasswd
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "Install Docker" >> /var/tmp/bootstrap.txt
+                apt --yes install apt-transport-https ca-certificates curl software-properties-common
+                curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+                add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"
+                apt update
+                apt-cache policy docker-ce
+                apt --yes install docker-ce
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "grant mastodon user docker" >> /var/tmp/bootstrap.txt
+                usermod -aG docker mastodon
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "install docker-compose" >> /var/tmp/bootstrap.txt
+                curl -L https://github.com/docker/compose/releases/download/1.22.0/docker-compose-$(uname -s)-$(uname -m) -o /usr/local/bin/docker-compose
+                chmod +x /usr/local/bin/docker-compose
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "Activate Swap" >> /var/tmp/bootstrap.txt
+                fallocate -l 5G /swapfile
```

Comment:
A configurable swapfile size parameter would be nice. Depending on the scale of the box, this may or may not be wanted

Reply:
A lot of there Mastodon instances are running on 1-2GB instances.. so 5GB was overkill..

but I've turned that into a parameter.